### PR TITLE
[OpenAl] fix link bug

### LINKED ISSUE #1528: [Openal-soft] can't link because latomic is not linked
error link latomic missing
```
-ldl -pthread /usr/lib/x86_64-linux-gnu/libasound.so /usr/lib/x86_64-linux-gnu/libpulse.so /usr/lib/x86_64-linux-gnu/libdbus-1.so /usr/lib/x86_64-linux-gnu/libSDL2.so /usr/lib/x86_64-linux-gnu/libsndio.so -Wl,--end-group
/usr/bin/ld: subprojects/openal-soft-1.23.1/libopenal.so.1.23.1.p/core_context.cpp.o: in function `std::atomic<std::bitset<3ul> >::is_lock_free() const':
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/atomic:251: undefined reference to `__atomic_is_lock_free'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

i have juste make a sfml.wrap project and openal-soft wrap

### PR DESCRIPTION
**Pull request by nda-cunh**
close #1528

### CODE REVIEW

**Comment by nda-cunh** [comment]
I don't know github, how to draft a Merge Request ? :) thank

**Comment by benoit-pierre** [comment]
Top right, below "Reviewers".

File: subprojects/packagefiles/openal-soft/meson.build [hunk 1]
```diff
@@ -6,9 +6,20 @@ project(
...
+
 cpp = meson.get_compiler('cpp')
+
+# Some systems may need libatomic for atomic functions to work
+if cpp.compiles('''#include <atomic>
+std::atomic<int> foo{0};
+int main() { return foo.fetch_add(2); }''')
+  atomic_dep = cpp.find_library('atomic', required: true)
+  openal_all_dependencies += atomic_dep
+endif
```
Comment (benoit-pierre) [inline subprojects/packagefiles/openal-soft/meson.build:20]: You can use `dependency('', required: false)` as a fallback instead:
```suggestion
# Some systems may need libatomic for atomic functions to work
atomic_dep = dependency('', required: false)
if cpp.compiles('''#include <atomic>
std::atomic<int> foo{0};
int main() { return foo.fetch_add(2); }''')
  atomic_dep = cpp.find_library('atomic', required: true)
endif
```
And remove all your `openal_all_dependencies` changes and unconditionally add `atomic_dep` to `openal_lib`'s dependencies.

**Comment by benoit-pierre** [comment]
Don't forget to bump the revision in `releases.json`.

**Comment by benoit-pierre** [comment]
```diff
--- i/releases.json
+++ w/releases.json
@@ -2488,6 +2488,7 @@
       "openal"
     ],
     "versions": [
+      "1.23.1-2",
       "1.23.1-1",
       "1.23.0-1",
       "1.22.2-8",
```

**Comment by neheb** [comment]
Alternate solution: https://github.com/mesonbuild/wrapdb/blob/master/subprojects/packagefiles/taglib/taglib/meson.build#L1

**Comment by neheb** [comment]
Alternative was merged.

**Pull request closed without merging**